ledger/narwhal/transmission-id/src/transmission_type.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO (raychu86): Move this implementation.
+
+use super::*;
+
+/// The transmission type.
+#[derive(Clone, Copy, PartialEq, Eq, Hash)]
+pub enum TransmissionType {
+    Ratification = 0,
+    Transaction = 1,
+    Solution = 2,
+}
+
+impl TransmissionType {
+    /// Returns the transmission type name.
+    pub fn type_name(&self) -> &str {
+        match self {
+            Self::Ratification => "ratification",
+            Self::Transaction => "transaction",
+            Self::Solution => "solution",
+        }
+    }
+
+    /// Returns the transmission type for the given numeric identifier.
+    pub fn from(type_id: u8) -> Result<Self> {
```

Comment:
nit: this should instead be an `impl TryFrom<u8> for TransmissionType`; `From` conversions are infallible